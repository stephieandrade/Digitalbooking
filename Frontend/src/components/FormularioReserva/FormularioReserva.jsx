import React, { useState, useEffect, useContext } from "react";
import "./formulario-reserva.scoped.css";
import { Contexto } from "../Contexto/Contexto";

export default function FormularioReserva({ nombre, apellido, email }) {
  //constantes de mensaje y estado para el error al no ingresar ciudad y para setear ciudad
  const { estado, setEstado } = useContext(Contexto);

  //funcion para setear la ciudad en el contexto
  function handleCiudadChange(e) {
    setEstado({
      ...estado,
      reserva: { ...estado.reserva, ciudad: e.target.value },
    });
  }

  return (
    <div className="contenedorFormulario">
      <form>
        <div className="formulario1">
          <div className="tituloinput">
            <h3>Nombre</h3>
            <input
              className="nombreInput"
              type="text"
              value={estado.session.nombre}
              name="nombre"
              placeholder="Nombre"
              disabled={true}
            />
          </div>
          <div className="tituloinput">
            <h3>Apellido</h3>
            <input
              className="apellidoInput"
              type="text"
              value={estado.session.apellido}
              name="apellido"
              placeholder="Apellido"
              disabled={true}
            />
          </div>
        </div>
        <div className="formulario2">
          <div className="tituloinput">
            <h3>Correo electr√≥nico</h3>
            <input
              className="emailInput"
              type="email"
              name="email"
              value={estado.session.sub}
              placeholder="Correo electronico "
              disabled={true}
            />
          </div>
          <div className="tituloinput">
            <h3>Ciudad</h3>
            <input
              className="ciudadInput"
              type="text"
              //npvalue={ciudad}
              name="ciudad"
              placeholder="Ciudad"
              onChange={handleCiudadChange}
              required={true}
            />
            {
              <div
                className={`errorMsg ${
                  estado.ciudadError === "" ? "hidden" : ""
                }`}
              >
                <p className="hiddenP">{estado.ciudadError}</p>
              </div>
            }
          </div>
        </div>
      </form>
    </div>
  );
}
