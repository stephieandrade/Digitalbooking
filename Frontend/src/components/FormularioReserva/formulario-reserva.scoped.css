div.contenedorFormulario {
    background: #FFFFFF;
    border: 1px solid #DFE4EA;
    box-sizing: border-box;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    border-radius: 8px;
    padding: 18px;
    margin-left: 13px;
    margin-right: 13px;
}


.formulario1 input,
.formulario2 input {
    width: 100%;
    background: #DFE4EA;
    height: 40px;
    box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.15);
    border-radius: 5px;
    outline: none;
}

.nombreInput::placeholder {
    font-weight: 500;
    font-size: 12px;
    line-height: 14px;
    opacity: 0.75;
   
}
.apellidoInput::placeholder {
    font-weight: 500;
    font-size: 12px;
    line-height: 14px;
    opacity: 0.75;
    
}
.emailInput::placeholder {
    font-weight: 500;
    font-size: 12px;
    line-height: 14px;
    opacity: 0.75;
   
} 

.ciudadInput::placeholder{
    font-weight: 500;
    font-size: 12px;
    line-height: 14px;
    opacity: 0.75;
}



.hidden {
    visibility: hidden;
  }

.hiddenP{
    color: red;
}

.tituloinput{
   padding-bottom: 2%;    
}

input.ciudadInput{
    padding-left: 12px;
}
input.nombreInput{
    padding-left: 12px;
}
input.apellidoInput{
    padding-left: 12px;
}
input.emailInput{
    padding-left: 12px;
}


@media (min-width: 768px) {
    .contenedorFormulario {
        display: flex;
        flex-direction: column;
        width: 96%;
    }

    .formulario1,
    .formulario2 {
        flex-direction: row;
        gap: 22px;
        justify-content: space-between;

    }

    .formulario1 label {
        width: 100%;
    }

    .formulario2 label {
        width: 100%;
        padding-top: 20px;

    }
}

@media (min-width: 992px) {
    .contenedorFormulario {
        display: flex;
        justify-content: space-around;
        width: 96%;
        height: 217px;
    }
    div.formulario1{
        display: flex;
        flex-direction: row;
    }
    div.formulario2{
        display: flex;
        flex-direction: row;
        padding-top: 3%;
    }
    .tituloinput {
    width: 100%;
    padding-bottom: 0;
    }
}