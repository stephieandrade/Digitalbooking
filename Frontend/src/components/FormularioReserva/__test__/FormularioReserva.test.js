import React from "react";
import { shallow } from "enzyme";
import FormularioReserva from "../FormularioReserva"

describe('FormularioReserva Component', () =>{

    it("deber√≠a mostrar <FormularioReserva/> correctamente ", () => {

        const wrapper = shallow(<FormularioReserva/>);
        expect(wrapper).toMatchSnapshot();
        });

    it ('tiene 4 etiquetas h3', () => {
        const wrapper = shallow(<FormularioReserva/>);
        const node = wrapper.find('h3');
        expect(node.length).toEqual(4);
       });

    it ('tiene 4 input', () => {
        const wrapper = shallow(<FormularioReserva/>);
        const node = wrapper.find('input');
        expect(node.length).toEqual(4);
       });
})
