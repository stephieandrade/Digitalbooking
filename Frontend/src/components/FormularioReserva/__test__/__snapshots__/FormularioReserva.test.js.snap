// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormularioReserva Component debería mostrar <FormularioReserva/> correctamente  1`] = `
<div
  className="contenedorFormulario"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="formulario1"
    >
      <div
        className="tituloinput"
      >
        <h3>
          Nombre
        </h3>
        <input
          disabled={true}
          name="nombre"
          placeholder="Nombre"
          type="text"
        />
      </div>
      <div
        className="tituloinput"
      >
        <h3>
          Apellido
        </h3>
        <input
          disabled={true}
          name="apellido"
          placeholder="Apellido"
          type="text"
        />
      </div>
    </div>
    <div
      className="formulario2"
    >
      <div
        className="tituloinput"
      >
        <h3>
          Correo electrónico
        </h3>
        <input
          disabled={true}
          name="email"
          placeholder="Correo electronico "
          type="email"
        />
      </div>
      <div
        className="tituloinput"
      >
        <h3>
          Ciudad
        </h3>
        <input
          name="ciudad"
          onChange={[Function]}
          placeholder="Ciudad"
          type="text"
          value=""
        />
      </div>
    </div>
  </form>
</div>
`;
